server/controllers/configurationsController/index.js

```diff
@@ -57,50 +55,51 @@ const ConfigurationsController = {
     }
   },
   index: async (req, res) => {
-    try {
-      const dataSet = new Set()
-      const { appDb } = req.app.locals
-      const { uid } = req.params
-      const streamConfigurations = await ConfigModel.index(appDb, uid)
-
-      streamConfigurations.on('data', async (config) => {
-        streamConfigurations.pause()
-
-        const isUserCached = cachedUsers.has(config.owner)
-
-        if (isUserCached) {
-          const configOwner = cachedUsers.get(config.owner)
-
-          dataSet.add({ ...config, owner_display_name: configOwner })
-        } else {
-          const configOwner = await UserModel.findOne(
-            appDb,
-            { uid: config.owner },
-            { display_name: 1 }
-          )
-
-          cachedUsers.set(config.owner, configOwner.display_name)
-          dataSet.add({
-            ...config,
-            owner_display_name: configOwner.display_name,
-          })
-        }
-
-        streamConfigurations.resume()
-      })
-
-      await new Promise((resolve, reject) => {
-        streamConfigurations.on('end', () => resolve())
-
-        streamConfigurations.on('error', (err) => reject(err))
-      })
-
-      const data = Array.from(dataSet)
-
-      return res.status(200).json({ data })
-    } catch (error) {
-      return res.status(400).json({ error: error.message })
-    }
+    const cachedUsers = new Map()
+    const dataSet = new Set()
+    const { appDb } = req.app.locals
+    const { uid } = req.params
+    const isActive =
+      Object.hasOwn(req.query, 'status') && req.query.status === 'active'
+    const streamConfigurations = isActive
+      ? await ConfigModel.active(appDb, uid)
+      : await ConfigModel.all(appDb, uid)
+
+    streamConfigurations.on('data', async (config) => {
+      streamConfigurations.pause()
+
+      const isUserCached = cachedUsers.has(config.owner)
+
+      if (isUserCached) {
+        const configOwner = cachedUsers.get(config.owner)
+
+        dataSet.add({ ...config, owner_display_name: configOwner })
```

Comment:
I don't think a set is the right thing to use here, we're not getting any benefit using it.

<img width="205" alt="Screenshot 2024-05-09 at 10 45 31 AM" src="https://github.com/AMP-SCZ/dpdash/assets/2905145/1b94a4ed-7dc7-437d-a8ba-4d87950e8cf0">